### ISSUE TITLE
introduce/change/edit macro enable/disable AVX support, ensuring support for older intel macs

### ISSUE THREAD

**Issue by i3roly** [i3roly/firefox-dynasty#3]
currently there doesn't seem to be an easy way to do this.
using the mozilla toolchain, which downloaded:

- clang-dist-toolchain.tar.xz (macos14.4sdk)
- rustc-dist-toolchain.tar.xz (rust 1.78)
- 0b316ff0b317a9ce-minidump-stackwalk.tar.zst	
- 88e566424540d332-clang-tidy.tar.zst		
- 8fcbf88b085c12f6-cbindgen.tar.zst		
- b60e9e684c3cc7d2-clang-dist-toolchain.tar.xz	
- d6e7c31fd21ee5f1-dump_syms.tar.zst		
- f3a013ef384246ab-rustc-dist-toolchain.tar.xz
- 7bf993af34cfe2a5-nasm.tar.zst			
- 8f5df5e1768e8d5c-sysroot-wasm32-wasi.tar.zst	
- a63be58566319021-fix-stacks.tar.zst		
- d54b96b2c46ecb95-clang.tar.zst			
- deafcb9de8091018-node.tar.zst			
- fbd9e64dfaaf4d5a-sccache.tar.zst

we cannot:

1. simply edit-out the avx2 macros in build/moz.configure/toolchain.configure (it will still build with avx in the directories mentioned below)
2. use the negative (i.e. 'mno-avx' instead of 'mavx') since the moz.build files are not receptive to either flag (causing build errors in the mno-avx case).

and we must therefore focus on the moz.build files. 

from what i have seen in the third_party directory, there is hard-coding of
```
DEFINES["WEBRTC_ENABLE_AVX2"] = True
```

so let's see if there is a way to follow this macro up to the main build files and change it. 

to summarise, we need to focus on these subdirectories first
- media
- third_party

and systematise the avx2 invocation. the newer toolchain just assumes i have avx support (even though my mac does not) because intel macs without avx2 are officially supported past brojave.

**Comment by i3roly** [comment]
turns out it wasn't AVX compatibility, but rather they nuked a great commit by a contributor named glandium:

https://hg.mozilla.org/mozilla-central/rev/02dd305e8ba9

restoring this commit fixes the illegal instruction and should work all the way back to 10.9

**Comment by i3roly** [comment]
turns out it wasn't AVX compatibility, but rather they nuked a great commit by a contributor named glandium:

https://hg.mozilla.org/mozilla-central/rev/02dd305e8ba9

restoring this commit fixes the illegal instruction and should work all the way back to 10.9
